source/developers-guide/example-plugin/index.md

```diff
@@ -28,187 +26,201 @@ This example gives answers to the following questions:
+        $ids = $this->getCategoryIds($ids, $context);
 
-### Enlight_Controller_Action_PostDispatchSecure_Frontend/Widgets
-```php
-public function addTemplateDir()
-{
-    Shopware()->Container()->get('template')->addTemplateDir($this->Path() . 'Views/');
+        //now select all category data for the selected ids
+        $categories = $this->categoryService->getList($ids, $context);
+
+        $result = [];
+        foreach ($ids as $productId => $categoryId) {
+            if (!isset($categories[$categoryId])) {
+                continue;
+            }
+            $result[$productId] = $categories[$categoryId];
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param $ids
+     * @param $context
+     * @return array
+     */
+    private function getCategoryIds($ids, ShopContextInterface $context)
+    {
+        $query = $this->connection->createQueryBuilder();
+        $query->select(['seoCategories.article_id', 'seoCategories.category_id'])
+            ->from('s_articles_categories_seo', 'seoCategories')
+            ->andWhere('seoCategories.article_id IN (:productIds)')
+            ->andWhere('seoCategories.shop_id = :shopId')
+            ->setParameter(':shopId', $context->getShop()->getId())
+            ->setParameter(':productIds', $ids, Connection::PARAM_INT_ARRAY);
+
+        return $query->execute()->fetchAll(\PDO::FETCH_KEY_PAIR);
+    }
 }
 ```
-Registers the plugin's `Views` directory as a template directory for Shopware.
-Attention: This event listener listens to the global frontend and widgets post dispatch secure event. The plugin shouldn't do some performance sensitive tasks here, otherwise each post dispatch event in the store front will be slowed down.
 
-<div class="alert alert-info">
-<strong>Always register your template directory first</strong>
-<br>
-The plugin template directory should always be registered, even if it's not used in the current controller context. If your registration of the template depends on certain conditions, a Smarty Security error may occur. This could be confusing and annoying for other third-party developers, because it's not obvious where the error came from in the first place.
-</div>
+After plugin installation the service is available in the service container:
+For example:
 
-### Theme_Compiler_Collect_Plugin_Less
 ```php
-public function onCollectLessFiles()
-{
-    return new \Shopware\Components\Theme\LessDefinition(
-        [],
-        [__DIR__ . '/Views/frontend/_public/src/less/all.less']
-    );
-}
+$seoService = $this->container->get('shopware_storefront.seo_category_service');
 ```
-Adds the plugin `all.less` file to the Shopware LESS compiling step. This allows the plugin to implement custom frontend styling via LESS.
 
-## Plugin ListProductService
+## Theme_Compiler_Collect_Plugin_Less
+
+Adds the plugin `all.less` file to the `Resousrces/views/frontend/less` directory. If the plugin installed shopware loads this file automatically. 
```

Comment:
Add the all.less file to the Resources/views/frontend/less directory. If the plugin is installed and active, Shopware loads this file automatically

Reply:
d